buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/CopyRestApiTask.java

```diff
@@ -48,8 +48,9 @@
     private static final String REST_TEST_PREFIX = "rest-api-spec/test";
     private final ListProperty<String> includeCore = getProject().getObjects().listProperty(String.class);
     private final ListProperty<String> includeXpack = getProject().getObjects().listProperty(String.class);
+    private final DirectoryProperty outputResourceDir = getProject().getObjects().directoryProperty();
```

Comment:
Again we should assign this from an injected ObjectFactory in the constructor. Lets clean up the other usages of getProject() here too